### ISSUE TITLE
Better handle shorting of work item friendly name

### ISSUE THREAD

**Issue by ulisesh** [dotnet/arcade#11079]
The helix API lets you upload workitems with a friendly name longer than 200 characters but in the backend we just keep the first 200 characters. Later the API crashes when people query using the full name

**Comment by ChadNedzlek** [comment]
This isn't going to be a cheap fix.  We accepts workitem names, and pass them to our backends in a LOT of places.  The easiest thing to do would be to reject them, but that would break a lot of existing builds, so isn't practical.

I'm not 100% sure on the ROI for this work.

**Comment by jaredpar** [comment]
> I'm not 100% sure on the ROI for this work.

How are consumers supposed to deal with this? We're getting no feedback uploading items and then immediately getting errors trying to access them. There is no public API for shortening names, no constant saying how long a name can be, etc ... Bascially as a consumer how can we reliably access Helix data? 

This isn't a theoretical problem, it's caused runfo to drop lots of data over the last month (at least)

**Comment by ChadNedzlek** [comment]
Only a single repository has this problem. In general consumers don't generate such long workitem names, so most customers don't have to deal with this.   This behavior has existed since the very beginning of helix and hasn't caused a problem for the majority of users, and changing this is incredibly expensive and error prone.

**Comment by riarenas** [comment]
If only one repo is sending 200+ friendly work item names, could we start by patching the mechanism that particular repo uses to reject anything longer than 200 chars?

**Comment by alexperovich** [comment]
I don't believe this is so expensive. We have a bunch of apis, all of them documented. We can do the proper truncation at the beginning of each of them and that would fix this problem. We could even reuse the same code that truncates when processing the job list.

**Comment by jaredpar** [comment]
> We can do the proper truncation at the beginning of each of them and that would fix this problem

What is the truncation algorithm? Short term I could add it to runfo and unblock everything. I tried to reverse engineer it based on convos 

```csharp
            if (id.Length > 200)
            {
                id = id.Substring(0, 200);
                id = id + "...";
            }
```

This does not work though. The `DetailsAsync` API no longer fails but the data it returns does not appear to be valid. It's a `WorkItemDetails` instance with virtually everything set to `null`. 

<img width="609" alt="image" src="https://user-images.githubusercontent.com/146967/193363178-5597dc8c-aaaf-45eb-b4b3-648ed548824a.png">

If I had to guess based on behavior I'd say that when given a name it doesn't recognize it assumes it's a yet to be scheduled work item and returns data representing that.

**Comment by garath** [comment]
It's handled by the [TruncateString](https://dev.azure.com/dnceng/internal/_git/dotnet-helix-service?path=/src/HelixMonitoringSql/Helix.Utilities.Sql/SqlHelper.cs&version=GBmain&line=113&lineEnd=135&lineStartColumn=1&lineEndColumn=1&lineStyle=plain&_a=contents) method. So for a string limit of 200 it'll land at `value.Substring(0, size - 3) + "...";`

**Comment by jaredpar** [comment]
@garath thanks and yes that indeed fixes the problem. A very simple 

```csharp
if (id.Length > 200)
{
    id = id.Substring(0, 197);
    id = id + "...";
}
``` 

Fixes the problems here. This could easily be added to the Helix client libraries as well. It's a trivial change that causes the query behavior to match the upload behavior. It is extremely hard to see how this is "not a good ROI".

**Comment by adiaaida** [comment]
We have triaged this into https://github.com/dotnet/arcade/issues/6948.
